uxarray/grid/geometry.py

```diff
@@ -429,3 +430,154 @@ def _classify_polygon_location(face_edge_cart):
+            typically between -180 and 180 degrees, or 0 and 360 degrees.
+            Here, the longitude does not "wrap around" when it reaches the end of this range.
+
+    Periodic Longitude: In this case, the longitude is considered to wrap around the globe.
+            This means that if you have a longitude range from 350 to 10 degrees,
+            it is understood to cross the 0-degree meridian and actually represents a 20-degree span
+            (350 to 360 degrees, then 0 to 10 degrees).
+
+    Parameters
+    ----------
+    latlonbox_rad : np.ndarray
+        A latitude-longitude box represented by a 2x2 array in radians and lon0 represent the "left" side of the box.
+        while lon1 represent the "right" side of the box:
+        [[lat_0, lat_1], [lon_0, lon_1]].
+
+    Returns
+    -------
+    float
+        The width of the latitude-longitude box in radians.
+
+    Raises
+    ------
+    Exception
+        If the input longitude range is invalid.
+
+    Warning
+        If the input longitude range is flagged as periodic but in the form [lon0, lon1] where lon0 < lon1.
+        The function will automatically use the is_lon_periodic=False instead.
+    """
+
+    lon0, lon1 = latlonbox_rad[1]
+
+    # Check longitude range validity
+    if (lon0 < 0.0 or lon0 > 2.0 * np.pi) and lon0 != INT_FILL_VALUE:
+        raise Exception("lon0 out of range ({} not in [0, 2π])".format(lon0))
+
+    if lon0 <= lon1:
+        return lon1 - lon0
+    else:
+        # Adjust for periodicity
+        return 2 * np.pi - lon0 + lon1
+
+
+def _insert_pt_in_latlonbox(old_box, new_pt, is_lon_periodic=True):
+    """Update the latitude-longitude box to include a new point in radians.
+
+    This function compares the new point's latitude and longitude with the
+    existing latitude-longitude box and updates the box if necessary to include the new point.
+
+    Parameters
+    ----------
+    old_box : np.ndarray
+        The original latitude-longitude box in radian, a 2x2 array: [min_lat, max_lat],[left_lon, right_lon]].
+    new_pt : np.ndarray
+        The new latitude-longitude point in radian, an array: [lat, lon].
+    is_lon_periodic : bool, optional
+        Flag indicating if the latitude-longitude box is periodic in longitude (default is True).
+
+    Returns
+    -------
+    np.ndarray
+        Updated latitude-longitude box including the new point in radians.
+
+    Raises
+    ------
+    Exception
+        If logic errors occur in the calculation process.
+
+    Examples
+    --------
+    >>> _insert_pt_in_latlonbox(np.array([[1.0, 2.0], [3.0, 4.0]]),np.array([1.5, 3.5]))
+    array([[1.0, 2.0], [3.0, 4.0]])
+    """
+    if np.all(new_pt == INT_FILL_VALUE):
+        return old_box
+
+    latlon_box = copy.deepcopy(old_box)  # Create a copy of the old box
+    latlon_box = np.array(
+        latlon_box, dtype=np.float64
+    )  # Cast to float64, otherwise the following update might fail
+
```

Comment:
We can combine these by using the numpy copy. Since these just contain points, it should work as expected 

https://numpy.org/doc/stable/reference/generated/numpy.copy.html

Reply:
I have changed to use `np.copy`, however, preserving the `    latlon_box = np.array(
        latlon_box, dtype=np.float64
    )  ` is still necessary. Otherwise, when the input points contain the `INT_FILL_VALUE`, error will occur